internal/update/update.go

```diff
@@ -31,6 +33,54 @@ type StateEntry struct {
 	LatestRelease      ReleaseInfo `yaml:"latest_release"`
 }
 
+func ShouldCheckForExtensionUpdate() bool {
+	if os.Getenv("GH_NO_UPDATE_NOTIFIER") != "" {
+		return false
+	}
+	if os.Getenv("CODESPACES") != "" {
+		return false
+	}
+	return !IsCI() && IsTerminal(os.Stdout) && IsTerminal(os.Stderr)
+}
+
+func CheckForExtensionUpdate(em extensions.ExtensionManager, ext extensions.Extension, now time.Time) (*ReleaseInfo, error) {
+	if ext.IsLocal() {
+		return nil, nil
+	}
+
+	stateFilePath := filepath.Join(em.UpdateDir(ext.Name()), "state.yml")
+	stateEntry, _ := getStateEntry(stateFilePath)
+	if stateEntry != nil && now.Sub(stateEntry.CheckedForUpdateAt).Hours() < 24 {
+		return nil, nil
+	}
+
+	releaseInfo := &ReleaseInfo{
+		Version: ext.LatestVersion(),
+		URL:     ext.URL(),
+	}
+
+	err := setStateEntry(stateFilePath, now, *releaseInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	if ext.UpdateAvailable() {
+		return releaseInfo, nil
+	}
+
+	return nil, nil
+}
+
+func ShouldCheckForUpdate(updaterEnabled string) bool {
+	if os.Getenv("GH_NO_UPDATE_NOTIFIER") != "" {
+		return false
+	}
+	if os.Getenv("CODESPACES") != "" {
+		return false
+	}
+	return updaterEnabled != "" && !IsCI() && IsTerminal(os.Stdout) && IsTerminal(os.Stderr)
```

Comment:
I agree this is a little too obtuse as-is.

In retrospect after our paired review, I got to thinking whether the duplicate logic between the 2 functions was valuable.  I began thinking this specific criteria should be moved out of to the gh update logic so it could be reused.